lms/djangoapps/courseware/tests/test_video_mongo.py

```diff
@@ -1093,47 +1093,74 @@ def test_source_is_empty(self):
                 'youtube': None,
                 'desktop_mp4': 'https://mp4.com/dm.mp4'
             },
-            ['https://hls.com/hls.m3u8']
+            ['https://www.youtube.com/watch?v=3_yD_cEKoCk']
         ),
         (
             {
-                'desktop_webm': 'https://webm.com/dw.webm',
+                'desktop_webm': None,
                 'hls': None,
                 'youtube': None,
+                'desktop_mp4': None
+            },
+            ['https://www.youtube.com/watch?v=3_yD_cEKoCk']
+        ),
+    )
+    @ddt.unpack
+    @patch('xmodule.video_module.video_module.HLSPlaybackEnabledFlag.feature_enabled', Mock(return_value=True))
+    def test_val_encoding_in_context(self, val_video_encodings, video_url):
+        """
+        Tests that the val encodings correctly override the video url when the edx video id is set and
+        one or more encodings are present.
+        """
+        with patch('xmodule.video_module.video_module.edxval_api.get_urls_for_profiles') as get_urls_for_profiles:
```

Comment:
can we do somthing like?
`with patch('xmodule.video_module.video_module.edxval_api.get_urls_for_profiles', return_value= val_video_encodings):`

Reply:
I don't think we can do like this. ddt is unpacked as positional arguments passed to function. `val_video_encodings ` is not known outside the function scope.